### PR TITLE
feat: add a `skipAutoLaunch` option

### PR DESCRIPTION
**Pull request by KentoNishi**
The auto-launching behavior can be undesirable in many circumstances like:

- using alternative browsers like edge (my setup)
- using WSL (also my setup)

I've added a `skipAutoInstall` option to easily disable auto-installing and launching to deal with that :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,6 +180,8 @@ export default defineConfig({
 > Watch mode will not reload properly when the manifest changes. You'll need to restart the `vite build --watch` command use the updated manifest.
 >
 > This is a limitation of [`web-ext`](https://www.npmjs.com/package/web-ext)
+>
+> Set `skipAutoInstall` to `true` to skip the automatic installation of the extension.
```
Comment (aklinker1) [inline README.md:184]:
```suggestion

Set `skipAutoInstall` to `true` to skip the automatic installation of the extension.
```

File: lib/index.ts [hunk 2]
```diff
@@ -55,6 +55,11 @@ interface BrowserExtensionPluginOptions {
    */
   webExtConfig?: any;
 
+  /**
+   * Used to disable auto-installing the extension when in watch mode. Default value is `false`.
+   */
+  skipAutoInstall?: boolean;
```
Comment (aklinker1) [inline lib/index.ts:61]: Can we rename this to `disableAutoLaunch`?

**Review by aklinker1 — CHANGES_REQUESTED** [review]
Code looks good. Couple of changes and I'll get this merged and released

Could you elaborate on the two use cases you mentioned? For edge, you should be able to point `web-ext`'s `chromiumBinary` to edge's path, and it should auto-install just fine if that's the problem:

```ts
WebExtension({
  matnifest: ...,
  assets: ...,
  webExtConfig: {
    chromiumBinary: "/c/path/to/edge.exe",
  },
});
```

For WSL, I know there are limitations when interacting with windows programs (I've had problems with electron in the past), but what's the limitation here?

**Comment by KentoNishi** [comment]
> For edge, you should be able to point web-ext's chromiumBinary to edge's path, and it should auto-install just fine if that's the problem

Yeah that would work I think, but I often have an inspected extension page open before starting the dev server, and it bugs me that the tab gets closed when the extension reloads. I prefer reloading manually when I want to, but that's just my preference lol

> For WSL, I know there are limitations when interacting with windows programs (I've had problems with electron in the past), but what's the limitation here?

When the package is being loaded, it must be loaded from `\\wsl$\tmp` rather than `/tmp` because the `/tmp` bath doesn't exist on the `c` drive. WSL2's inter-OS operations are also very very slow, so I'd rather avoid driving chromium from WSL entirely.

I'll get to work on the small fixes now :D

**Comment by KentoNishi** [comment]
@aklinker1By the way, I just wanted to say thank you for making such a great plugin. I'm kinda shocked how much effort you've put into a niche project like this, hopefully it becomes popular as vite gains more traction 🚀

**Review by aklinker1 — APPROVED** [review]

**Comment by aklinker1** [comment]
@KentoNishi Thanks! Bundling a web extension from scratch is difficult to configure, so I decided to publish my build process I made for one of my personal extensions.

This has been published in [v1.1.0](https://github.com/aklinker1/vite-plugin-web-extension/releases/tag/v1.1.0) BTW

**Pull request merged**